src/proc/y16i-to-y16y16.cpp

```diff
@@ -0,0 +1,42 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include "y16i-to-y16y16.h"
+#include "stream.h"
+// CUDA TODO
+//#ifdef RS2_USE_CUDA
+//#include "cuda/cuda-conversion.cuh"
+//#endif
+
+namespace librealsense
+{
+    struct y16i_pixel { uint16_t left : 16, right : 16;
+                        // left and right data are shifted 6 times so that the MSB 10 bit will hold the data
+                        uint16_t l() const { return left << 6; }
+                        uint16_t r() const { return right << 6; } };
+    void unpack_y16_y16_from_y16i(byte* const dest[], const byte* source, int width, int height, int actual_size)
```

Comment:
Let's rename the function to `unpack_y10msb_y10msb_from_y16i` to allow in future to use real 16but data (if becomes available)